File: src/windows/magma/app/compile/value/ConstructionCaller.c
#include "./ConstructionCaller.h"
export class ConstructionCaller implements Caller {
	&[I8] right;
	Platform platform;
	constructor (&[I8] right, Platform platform) {
		this.right = right;
		this.platform = platform;
	}
}

&[I8] generate(Platform platform) {
	if (Platform/*auto*/.Magma === this/*auto*/.platform) {
		return this/*auto*/.right;
	}
	return "new " + this/*auto*/.right;
}
Option<Value> findChild() {
	return new None<Value>(/*auto*/);
}

File: src/java/magma/app/compile/define/ConstructionCaller.java
package magma.app.compile.define;

import magma.api.option.None;
import magma.api.option.Option;
import magma.app.compile.value.Caller;
import magma.app.compile.value.Value;

public record ConstructionCaller(String right) implements Caller {
    @Override
    public String generate() {
        return "new " + this.right;
    }

    @Override
    public Option<Value> findChild() {
        return new None<Value>();
    }
}
